iceoryx_posh/source/roudi/iceoryx_port_pool.cpp

```diff
@@ -77,26 +77,28 @@ IceOryxPortPool::addReceiverPort(const capro::ServiceDescription& serviceDescrip
 }
 
 /// @deprecated #25
-void IceOryxPortPool::removeSenderPort(SenderPortType::MemberType_t* const portData) noexcept
+void IceOryxPortPool::removeSenderPort(const SenderPortType::MemberType_t& portData) noexcept
 {
-    m_portPoolData->m_senderPortMembers.erase(portData);
```

Comment:
Ok, worked before because the iterators where pointers (for cxx::vector). Now to ensure deletion we need to provide a reference to an object actually in the container (equivalent to a pointer in this regard). Should be fine, but worth thinking about that it is still correct and equivalent to before. We cannot delete by id or name or anything else. The address of the object to be deleted is essentially the id.

Reply:
> We cannot delete by id or name or anything else. The address of the object to be deleted is essentially the id.

Q: Could there appear a race where the pointer-is-ID approach becomes corrupted due to different address ranges in processes... ?